sample_annotator/clients/submission_portal_client.py

```diff
@@ -0,0 +1,185 @@
+import pandas as pd
+import nmdc_schema.nmdc as nmdc
+import linkml
+import numpy as np
+import linkml_runtime.dumpers.json_dumper as dumper
+import requests
+import json
+from clients.nmdc.runtime_api_client import RuntimeApiSiteClient
+from dotenv import dotenv_values
+
+
+
+#I had initially misunderstood, and did not design the code as part of an class/object so these are mostly
+#independent functions that I've added to this overarching class
+
+#API things are still missing as I only just got access to them ~1 hour before pushing this code up
+class submission_portal_client:
+    
+    #params:
+    #mapping_path: file path to mappings between column names
+    #env_path: path to env file with API information (not used yet)
+    def __init__(self,env_path):
+        self.env_path = env_path
+
+        #API things
+        self.env_vars = dotenv_values(env_path)
+
+        self.api_client = RuntimeApiSiteClient(self.env_vars['BASE_URL'],self.env_vars['SITE_ID'],self.env_vars['CLIENT_ID'],self.env_vars['CLIENT_SECRET'])
+
+    
+    #params:
+    #submission_id: a submission_id for the NMDC data portal
+    #returns: a json type response containing the data of the NMDC submission
+    def get_submission_json(self,submission_id):
+        response = requests.request(
+            "GET",
+            f"https://data.microbiomedata.org/api/metadata_submission/{submission_id}",
+            cookies={"session": self.env_vars["DATA_PORTAL_COOKIE"]},
+        )
+        return response.json()
+
+
+    def mint_nmdc_ids(self,num_ids):
+        json_dic = {"schema_class": {"id": "nmdc:Activity"},"how_many": num_ids}
```

Comment:
Are you sure you want to be minting ids for the `nmdc:Activity` type? Isn't it `nmdc:Biosample` because you're trying to assign ids to each row from the Submission Portal interface, and each of those rows are biosamples?

Reply:
I was copying the example for minting IDs and hadnt realized there were multiple types. Ive made this change and it now mints biosample IDs.